src/generate.jl

```diff
@@ -1,135 +1,128 @@
+
+    for (i, loc, sym) in [(0, :below, :x), (1, :left, :y)]
+        @eval begin
+            tick_type, tickform_type = _axis_type(getfield($p, symbol($"$(sym)_axis_type")))
+
+            tick = Bokehjs.Ticker(tick_type)
+            tickform = Bokehjs.TickFormatter(tickform_type)
+            axis = Bokehjs.LinearAxis($i, tickform, tick, $bkplot)
+            grid = Bokehjs.Grid($i, $bkplot, tick)
+
+            for obj in (tick, axis, tickform, grid)
+                pushdict!($obs, obj, $doc)
+            end
+
+            push!($renderers, axis)
+            push!($renderers, grid)
+            $axes[symbol($"$loc")] = [axis]
+        end
+    end
 
     tools = Bokehjs.PlotObject[]
-    if in(:pan, plot.tools)
-        pantool = Bokehjs.Metatool("PanTool", bkplot, String["width", "height"])
+    if in(:pan, p.tools)
+        pantool = Bokehjs.Metatool("PanTool", bkplot,
+                                   AbstractString["width", "height"])
         pushdict!(obs, pantool, doc)
         push!(tools, pantool)
     end
-    if in(:wheelzoom, plot.tools)
-        wheelzoomtool = Bokehjs.Metatool("WheelZoomTool", bkplot, String["width", "height"])
+    if in(:wheelzoom, p.tools)
+        wheelzoomtool = Bokehjs.Metatool("WheelZoomTool", bkplot,
+                                         AbstractString["width", "height"])
         pushdict!(obs, wheelzoomtool, doc)
         push!(tools, wheelzoomtool)
     end
-    if in(:boxzoom, plot.tools)
+    if in(:boxzoom, p.tools)
         boxzoomtool = Bokehjs.Metatool("BoxZoomTool", bkplot)
         pushdict!(obs, boxzoomtool, doc)
         push!(tools, boxzoomtool)
     end
-    if in(:resize, plot.tools)
+    if in(:resize, p.tools)
         resizetool = Bokehjs.Metatool("ResizeTool", bkplot)
         pushdict!(obs, resizetool, doc)
         push!(tools, resizetool)
     end
-    if in(:reset, plot.tools)
+    if in(:reset, p.tools)
         resettool = Bokehjs.Metatool("ResetTool", bkplot)
         pushdict!(obs, resettool, doc)
         push!(tools, resettool)
     end
 
-    push!(renderers, axis0)
-    push!(renderers, axis1)
-    push!(renderers, grid0)
-    push!(renderers, grid1)
-
-    axes = @compat Dict{Symbol, Vector}(
-        :above => [],
-        :below => [axis0],
-        :left => [axis1],
-        :right => []
-    )
-
-    bkplot = Bokehjs.Plot(bkplot,
-                dr1x,
-                dr1y,
-                renderers,
-                axes,
-                tools,
-                plot.title,
-                plot.height,
-                plot.width)
+    bkplot = Bokehjs.Plot(bkplot, dr1x, dr1y, renderers, axes, tools, p.title,
+                          p.height, p.width)
     pushdict!(obs, bkplot, doc)
 
     plotcontext = Bokehjs.PlotContext(bkplot)
     pushdict!(obs, plotcontext, doc)
 
     indent = DEBUG ? 2 : 0
-    method_exists(json, (Dict, Int)) ? (json(obs, indent), plotcontext): 
+    method_exists(json, (Dict, Int)) ? (json(obs, indent), plotcontext):
```

Comment:
i think `method_exists` can go, JSON.jl for julia v0.4 definitely has indent support